src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,168 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
+    private static final String FORMAT_DATE_TIME = "dd MMMM yyyy HH:mm";
+    private static final String FORMAT_CUSTOM = "dd MMM yyy";
+    private static final String KYIV_TIME = "+02:00";
+
     /**
      * Return the current date as a String depending on a query.
-     *
+     * <p>
      * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
+     * - FULL - current date as a whole: year, month, day of month
+     * formatted as `YYYY-MM-DD` (a default return value);
+     * - YEAR - current year;
+     * - MONTH - name of the current month;
+     * - DAY - current day of month;
      * In any other case throw DateTimeException.
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        switch (datePart) {
+            case DAY:
+                return String.valueOf(LocalDate.now().getDayOfMonth());
+            case FULL:
+                return String.valueOf(LocalDate.now());
+            case YEAR:
+                return String.valueOf(LocalDate.now().getYear());
+            case MONTH:
+                return String.valueOf(LocalDate.now().getMonth());
+            default:
+                throw new DateTimeException("Incorrect input");
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/date-time-api#avoid-calling-localdatenow-multiple-times-in-one-method-lets-create-a-variable-for-this-purpose

Reply:
let's create a local variable for LocalDate.now();